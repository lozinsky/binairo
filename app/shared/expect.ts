import { assert } from '~/shared/assert';

export function expectToSatisfy<T>(value: unknown, predicate: (value: unknown) => value is T, message?: string) {
  assert(predicate(value), message ?? 'Expected value to satisfy the predicate');

  return value;
}

export function expectToBeDefined<T>(value: T, message?: string) {
  assert(value != null, message ?? 'Expected value to be defined');

  return value;
}

export function expectNotToBeNaN(value: number, message?: string) {
  assert(!isNaN(value), message ?? 'Expected value not to be "NaN"');

  return value;
}
