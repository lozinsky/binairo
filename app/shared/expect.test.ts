import { expect, test } from 'vitest';

import { expectNotToBeNaN, expectToBeDefined, expectToSatisfy } from './expect';

test('returns value and does not throw if value satisfies predicate', () => {
  expect(expectToSatisfy('value', (value): value is string => value === 'value')).toMatchSnapshot();
});

test('does not return value and throws if value does not satisfy predicate', () => {
  expect(() => {
    expectToSatisfy('value', (value): value is string => value === 'other');
  }).toThrowErrorMatchingSnapshot();
});

test('does not return value and throws if value does not satisfy predicate with message', () => {
  expect(() => {
    expectToSatisfy('value', (value): value is string => value === 'other', 'Message');
  }).toThrowErrorMatchingSnapshot();
});

test('returns value and does not throw if value is not "null" or "undefined"', () => {
  expect(expectToBeDefined('value')).toMatchSnapshot();
});

test('returns value and does not throw if value is not "null" or "undefined"', () => {
  expect(expectToBeDefined(0)).toMatchSnapshot();
});

test('returns value and does not throw if value is not "null" or "undefined"', () => {
  expect(expectToBeDefined(false)).toMatchSnapshot();
});

test('does not return value and throws if value is "null"', () => {
  expect(() => {
    expectToBeDefined(null);
  }).toThrowErrorMatchingSnapshot();
});

test('does not return value and throws if value is "null" with message', () => {
  expect(() => {
    expectToBeDefined(null, 'Message');
  }).toThrowErrorMatchingSnapshot();
});

test('does not return value and throws if value is "undefined"', () => {
  expect(() => {
    expectToBeDefined(undefined);
  }).toThrowErrorMatchingSnapshot();
});

test('does not return value and throws if value is "undefined" with message', () => {
  expect(() => {
    expectToBeDefined(undefined, 'Message');
  }).toThrowErrorMatchingSnapshot();
});

test('returns value and does not throw if value is not "NaN"', () => {
  expect(expectNotToBeNaN(0)).toMatchSnapshot();
});

test('does not return value and throws if value is "NaN"', () => {
  expect(() => {
    expectNotToBeNaN(NaN);
  }).toThrowErrorMatchingSnapshot();
});

test('does not return value and throws if value is "NaN" with message', () => {
  expect(() => {
    expectNotToBeNaN(NaN, 'Message');
  }).toThrowErrorMatchingSnapshot();
});
