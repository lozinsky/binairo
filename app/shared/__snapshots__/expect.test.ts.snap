// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`does not return value and throws if value does not satisfy predicate 1`] = `[AssertionError: Expected value to satisfy the predicate]`;

exports[`does not return value and throws if value does not satisfy predicate with message 1`] = `[AssertionError: Message]`;

exports[`does not return value and throws if value is "NaN" 1`] = `[AssertionError: Expected value not to be "NaN"]`;

exports[`does not return value and throws if value is "NaN" with message 1`] = `[AssertionError: Message]`;

exports[`does not return value and throws if value is "null" 1`] = `[AssertionError: Expected value to be defined]`;

exports[`does not return value and throws if value is "null" with message 1`] = `[AssertionError: Message]`;

exports[`does not return value and throws if value is "undefined" 1`] = `[AssertionError: Expected value to be defined]`;

exports[`does not return value and throws if value is "undefined" with message 1`] = `[AssertionError: Message]`;

exports[`returns value and does not throw if value is not "NaN" 1`] = `0`;

exports[`returns value and does not throw if value is not "null" or "undefined" 1`] = `"value"`;

exports[`returns value and does not throw if value is not "null" or "undefined" 2`] = `0`;

exports[`returns value and does not throw if value is not "null" or "undefined" 3`] = `false`;

exports[`returns value and does not throw if value satisfies predicate 1`] = `"value"`;
